test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -0,0 +1,144 @@
+        this.serviceBuilder.initialize();
+        this.tableStore = this.serviceBuilder.createTableStoreService();
+        int port = TestUtils.getAvailableListenPort();
+        this.serverListener = new PravegaConnectionListener(false, port, mock(StreamSegmentStore.class), this.tableStore);
+        this.serverListener.startListening();
+
+        this.connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder().build());
+        this.controller = new MockController(ENDPOINT, port, this.connectionFactory, true);
+        this.controller.createScope(SCOPE);
+        this.keyValueTableFactory = new KeyValueTableFactoryImpl(SCOPE, this.controller, this.connectionFactory);
+    }
+
+    @After
+    public void tearDown() {
+        this.controller.close();
+        this.connectionFactory.close();
+        this.serverListener.close();
+        this.serviceBuilder.close();
+        ResourceLeakDetector.setLevel(this.originalLevel);
+    }
+
+    /**
+     * Smoke Test. Verify that the KeyValueTable can be created and deleted.
+     */
+    @Test
+    public void testCreateDeleteKeyValueTable() {
+        val kvt = newKeyValueTableName();
+        boolean created = this.controller.createKeyValueTable(kvt.getScope(), kvt.getKeyValueTableName(), DEFAULT_CONFIG).join();
+        Assert.assertTrue(created);
+        val segments = this.controller.getCurrentSegmentsForKeyValueTable(kvt.getScope(), kvt.getKeyValueTableName()).join();
+        Assert.assertEquals(DEFAULT_CONFIG.getPartitionCount(), segments.getSegments().size());
+        for (val s : segments.getSegments()) {
+            // We know there's nothing in these segments. But if the segments hadn't been created, then this will throw
+            // an exception.
+            this.tableStore.get(s.getScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT).join();
+        }
+
+        // Verify re-creation does not work.
+        Assert.assertFalse(this.controller.createKeyValueTable(kvt.getScope(), kvt.getKeyValueTableName(), DEFAULT_CONFIG).join());
+
+        // Delete and verify segments have been deleted too.
+        val deleted = this.controller.deleteKeyValueTable(kvt.getScope(), kvt.getKeyValueTableName()).join();
+        Assert.assertTrue(deleted);
+        Assert.assertFalse(this.controller.deleteKeyValueTable(kvt.getScope(), kvt.getKeyValueTableName()).join());
+        for (val s : segments.getSegments()) {
+            AssertExtensions.assertSuppliedFutureThrows(
+                    "Segment " + s + " has not been deleted.",
+                    () -> this.tableStore.get(s.getScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT),
+                    ex -> ex instanceof StreamSegmentNotExistsException);
+        }
+    }
+
+    @Test
+    @Override
+    @Ignore
+    public void testIterators() {
+        // TODO: iterators not yet supported server-side.
```

Comment:
#4656  has a branch with all the changes. I can't submit that PR until I get this one in since they clash on a few files - including this one.

We'll have to do them in sequence, and I chose to do this PR first.